ganttproject/src/main/java/net/sourceforge/ganttproject/GanttProject.java

```diff
@@ -148,15 +146,36 @@ public class GanttProject extends GanttProjectBase implements ResourceView, Gant
 
   private FXSearchUi mySearchUi;
 
-  /**
-   * Number of transactions committed by the Colloboque server.
-   */
-  private int myCommittedTxnNum = 0;
-  /**
-   * ID of the last transaction committed by the Colloboque server.
-   */
-  private String myLastCommittedTxnId = "";
-  private final Lock myXlogSendingLock = new ReentrantLock();
+  private static class TxnCommitInfo {
+    /** Base transaction ID received from the server. */
+    protected String baseTxnId = "";
+    /** Local ID of the last transaction committed by the server. */
+    protected int baseTxnLocalId = 0;
+
+    TxnCommitInfo() {}
+    TxnCommitInfo(String baseTxnId, int baseTxnLocalId) {
+      this.baseTxnId = baseTxnId;
+      this.baseTxnLocalId = baseTxnLocalId;
+    }
+  }
+
+  private static class TxnCommitInfoSync extends TxnCommitInfo {
+    synchronized Boolean compareAndSetBaseTxn(String expectedTxnId, String newTxnId) {
```

Comment:
You never check the return value of this method. Did you mean to add such code later?
Add TODO?

Reply:
Well, I think I don't want to check it
Calling it `set` wouldn't be correct. Maybe, it's better to rename it to `acceptCommittedTxn`?